wwwroot/js/timer.js

```diff
@@ -0,0 +1,55 @@
+﻿var seconds = 00;
+var tens = 00;
+var min = 00;
+var OutputMin = document.getElementById('min');
+var OutputSeconds = document.getElementById('second');
+var OutputTens = document.getElementById('tens');
+var buttonReset = document.getElementById('ms-reset');
+var interval;
+
+
+window.onload = function () {
+    clearInterval(interval);
+    interval = setInterval(startTime, 10);
+}
+
+//Needs stop function of clearInterval(interval); tied to win/lose condition
+
+buttonReset.addEventListener('click', () => {
+    clearInterval(interval);
+    tens = "00";
+    seconds = "00";
+    OutputSeconds.innerHTML = seconds;
+    OutputTens.innerHTML = tens;
+    console.log("reset button clicked");
+    interval = setInterval(startTime, 10);
+})
+
+function startTime() {
+    tens++;
+    if (tens <= 9) {
+        OutputTens.innerHTML = "0" + tens;
+    } else {
+        OutputTens.innerHTML = tens;
+    }
+
+    if (tens > 99) {
+        seconds++;
+        OutputSeconds.innerHTML = "0" + seconds;
+        tens = 0;
+        OutputTens.innerHTML = "0" + tens;
+    }
+
+    if (seconds > 9) {
+        OutputSeconds.innerHTML = seconds;
+    }
+
+    if (seconds > 59) {
+        min++;
+        OutputMin.innerHTML = "0" + min;
+        seconds = 0;
+        OutputSeconds.innerHTML = "0" + seconds;
+        tens = 0;
+        OutputTens.innerHTML = "0" + tens;
+    }
+}
```

Comment:
As it stands, the timer is currently reset on every interaction that causes the game board to be re-rendered. We should move the timer likely up a level to the Game `Index` and Start or Stop the timer within `board.js`. I'm assuming we can just bind a value within `timer.js` to the `window` and access it from `board.js`. Something like `window.$ms` might be fine.

---

I've moved this to its own file since `site.js` is loaded on every page and as a result will throw infinite console errors within the `setInterval` call since all the elements will be `null` if not on the game page. I also cleaned up some of the merge conflicts that popped up from the changes made in Joel's branch.